src/ActionEventEmitterEventStore.php

```diff
@@ -304,22 +307,62 @@ public function updateStreamMetadata(StreamName $streamName, array $newMetadata)
         }
     }
 
-    public function createQuery(): Query
+    public function createQuery(QueryFactory $factory = null): Query
     {
-        return $this->eventStore->createQuery();
+        if (null === $factory) {
+            $factory = $this->getDefaultQueryFactory();
+        }
+
+        return $factory->factory();
     }
 
-    public function createProjection(string $name, ProjectionOptions $options = null): Projection
-    {
-        return $this->eventStore->createProjection($name, $options);
+    public function createProjection(
+        string $name,
+        ProjectionOptions $options = null,
+        ProjectionFactory $factory = null
+    ): Projection {
+        if (null === $factory) {
+            $factory = $this->getDefaultProjectionFactory();
+        }
+
```

Comment:
shouldn't the ActionEventEmitterEventStore also invoke `$factory->setEventStore($this);`, not only when creating the default factory?